sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/io/AvroIO.java

```diff
@@ -326,6 +327,14 @@
       }
 
       @Override
+      public void populateDisplayData(DisplayData.Builder builder) {
+        builder.add("schema", type);
+        if (filepattern != null) {
+          builder.add("filePattern", filepattern);
+        }
```

Comment:
It looks like the `AvroSource` is created inside the `.apply()` method, after some validation. I'd like it to be possible to call `populateDisplayData()` in any context-- like `.toString()`.

Given that, do you think it makes sense to create an `AvroSource` inside of `populateDisplayData()` in order to let it register metadata, or let `Avro.Read.Bound` continue to register it directly?